daemon/images/image_list.go

```diff
@@ -97,6 +115,9 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 		if !beforeFilter.IsZero() && (img.Created == nil || !img.Created.Before(beforeFilter)) {
 			continue
 		}
+		if !untilFilter.IsZero() && (img.Created == nil || !img.Created.Before(untilFilter)) {
+			continue
+		}
```

Comment:
The `untilFilter` variable has identical semantics to `beforeFilter`: select images not created before that timestamp. Therefore the `untilFilter` variable is not needed. Update the `beforeFilter` variable instead.